js_modules/dagster-ui/packages/ui-core/src/search/SearchDialog.tsx

```diff
@@ -72,12 +72,46 @@ const initialState: State = {
 
 const DEBOUNCE_MSEC = 100;
 
+type SearchResultGroups = {
+  primaryResults: Fuse.FuseResult<SearchResult>[];
+  assetResults: Fuse.FuseResult<SearchResult>[];
+  assetFilterResults: Fuse.FuseResult<SearchResult>[];
+};
+
+function groupSearchResults(
+  primaryResults: Fuse.FuseResult<SearchResult>[],
+  secondaryResults: Fuse.FuseResult<SearchResult>[],
+  isAssetSearch: boolean,
+): SearchResultGroups {
+  if (isAssetSearch) {
+    return {
+      primaryResults: [],
+      assetResults: secondaryResults.filter(
+        (result) => result.item.type === SearchResultType.Asset,
+      ),
+      assetFilterResults: secondaryResults.filter((result) =>
+        isAssetFilterSearchResultType(result.item.type),
+      ),
+    };
+  } else {
+    return {
+      primaryResults,
+      assetResults: secondaryResults.filter(
+        (result) => result.item.type === SearchResultType.Asset,
+      ),
+      assetFilterResults: [],
+    };
+  }
+}
+
 export const SearchDialog = ({
   searchPlaceholder,
   isAssetSearch,
+  displayAsOverlay = true,
 }: {
-  searchPlaceholder: string;
+  searchPlaceholder?: string;
   isAssetSearch: boolean;
+  displayAsOverlay?: boolean;
```

Comment:
The addition of `isAssetSearch` is an indication that this isn't really the right component for this UI to use. I see that that's from another branch, so I'll take a look at that one.